pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -869,7 +869,7 @@ func (c *Cloud) InstanceID(name string) (string, error) {
 	}
 	inst, err := c.getInstanceByNodeName(name)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("getInstanceByNodeName failed for %q with %v", name, err)
```

Comment:
Yes, we have to be really careful with when we return InstanceNotFound.  According to the comments on cloudprovider.Instances, we must return InstanceNotFound, but actually that's wrong and it only applies to ExternalID (AFAICT), and you didn't wrap it.  I'll put in a separate PR to fix the comments.

To (try to) be clear: But what you're done here is right,  though it would be wrong if the docs were right.  At least I think so!